OrbitService/Utils.cpp

```diff
@@ -136,22 +139,30 @@ ErrorMessageOr<std::vector<ModuleInfo>> ParseMaps(std::string_view proc_maps_dat
   return result;
 }
 
-ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ReadTracepoints() {
+ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ListTracepoints() {
   std::vector<TracepointInfo> result;
 
-  for (const auto& category : fs::directory_iterator(kLinuxTracingEvents)) {
+  std::error_code error_code_category, error_code_name;
+
+  for (const auto& category : fs::directory_iterator(kLinuxTracingEvents, error_code_category)) {
     if (fs::is_directory(category)) {
-      for (const auto& name : fs::directory_iterator(category)) {
+      for (const auto& name : fs::directory_iterator(category, error_code_name)) {
         TracepointInfo tracepoint_info;
         tracepoint_info.set_name(fs::path(name).filename());
         tracepoint_info.set_category(fs::path(category).filename());
         result.emplace_back(tracepoint_info);
       }
     }
   }
+
   return result;
 }
 
+ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ReadTracepoints() {
+  OUTCOME_TRY(result, ListTracepoints());
```

Comment:
I might have caused some confusion here: OUTCOME_TRY is not catching any exceptions it is merely forwarding errors from "below" in a meaningful way. So there is no need to use it here. Also wrapping ListTracepoints in ReadTracepoints makes no sense. See above for a proper solution.